modules/post/windows/gather/psreadline_history.rb

```diff
@@ -0,0 +1,51 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/post/windows/user_profiles'
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info={})
+    super(update_info(info,
+        'Name' => 'Windows Gather PSReadline history',
+        'Description' => %q{
+          Gathers Power Shell history data from the target machine.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Garvit Dewan <d.garvit[at]gmail.com>' # @dgarvit
+        ],
+        'Platform' => %w{ win },
+        'SessionTypes' => [ 'meterpreter' ],
```

Comment:
Can you make this work on `shell`?

Reply:
How do I get all the users in a `shell` session? `Msf::Post::Windows::UserProfiles` doesn't seem to work in a `shell` session.

Reply:
I wouldn't bother. Most of the Post modules only support `meterpreter` and do not support `shell`.

```
# grep -rn SessionTypes modules/post/windows/gather/ | wc -l
117
# grep -rn SessionTypes modules/post/windows/gather/ | grep shell | wc -l
4
# grep -rn SessionTypes modules/post/windows/gather/ | grep meterp | wc -l
117
```